lib/task_helpers/exports/custom_buttons.rb

```diff
@@ -0,0 +1,56 @@
+module TaskHelpers
+  class Exports
+    class CustomButtons
+      class ExportArInstances
+        EXCLUDE_ATTRS = %w(id created_on updated_on created_at updated_at dialog_id resource_id).freeze
+        def self.export_object(obj, hash)
+          class_name = obj.class.name.underscore
+
+          $log.send(:info, "Exporting #{obj.class.name} [#{obj.try('name')}] with ID: #{obj&.id}")
+          (hash[class_name] ||= []) << item = { 'attributes' => build_attr_list(obj.try(:attributes)) }
+          create_association_list(obj, item)
+          descendant_list(obj, item)
+        end
+
+        def self.build_attr_list(attrs)
+          attrs&.except(*EXCLUDE_ATTRS)
+        end
+
+        def self.create_association_list(obj, item)
+          associations = obj.class.try(:reflections)
+          if associations
+            associations = associations.collect { |r| { r[0] => r[1].class.to_s.demodulize } }.select { |d| d.values.first != "BelongsToReflection" && d.keys.first != "all_relationships" }
```

Comment:
@d-m-u - Building the associations are in the weeds, but I wouldn't mind seeing variable names that reflect what the variable is e.g. `r`, `d`